typespec-extension/src/code-model-builder.ts

```diff
@@ -365,7 +365,7 @@ export class CodeModelBuilder {
                 // HTTP Authentication should use "Basic token" or "Bearer token"
                 schemeOrApiKeyPrefix = pascalCase(schemeOrApiKeyPrefix);
 
-                if (!(this.options.branded === false)) {
+                if (this.options["flavor"]?.toLocaleLowerCase() === "azure") {
```

Comment:
this may need be `!this.options["flavor"] || <existing>`

If no this `flavor`, it is still default to azure, at present.

Reply:
https://github.com/Azure/autorest.java/blob/a3e61b8ef7544e38785f0b01c480a324d3c646b5/typespec-extension/src/emitter.ts#L63
default is set here, so I did not do the check. I have also tested when flavor is not set, we will generate azure code. Also tested `flavor: xxx` (xxx is not Azure), it generates non-azure.
Let me know if you thought we should do double check.